test/miner_txn_mgr_SUITE.erl

```diff
@@ -309,12 +321,148 @@ txn_dependent_test(Config) ->
+                                                    ct:pal("no cached txns", []),
+                                                    false;
+                                                CachedTxnMap ->
+                                                    OUIs =
+                                                        maps:fold(fun(Txn, _TxnData, Acc) ->
+                                                                    TxnMod = blockchain_txn:type(Txn),
+                                                                    case TxnMod of
+                                                                        blockchain_txn_oui_v1 -> [Txn | Acc];
+                                                                        _ -> Acc
+                                                                    end
+                                                              end, [], CachedTxnMap),
+                                                    case length(OUIs) of
+                                                        2 -> true;
+                                                        _ -> false
+                                                    end
+                                            end
+                                        end, 60, 200),
+
+    ok = handle_get_cached_txn_result(Result, Miner, IgnoredTxns, Chain),
+
+    ok.
+
+txn_dup_payment_txn_test(Config) ->
+    %% send two payment txns with the same values
+    %% this is to test txn mgr does not dedup, the client is deliberately sending two txns with same data
+    %% in the case of payments, the dup txn WILL fail validations due to the invalid nonce
+    %% this will result in the first txn passing validation and the second failing
+    %% we want to ensure that only 1 of the payments is accepted/absorbed
+
+    Miner = hd(?config(non_consensus_miners, Config)),
+    ConMiners = ?config(consensus_miners, Config),
+    AddrList = ?config(tagged_miner_addresses, Config),
+
+    IgnoredTxns = [],
+    Addr = miner_ct_utils:node2addr(Miner, AddrList),
+
+    Chain = ct_rpc:call(Miner, blockchain_worker, blockchain, []),
+
+    PayerAddr = Addr,
+    Payee = hd(miner_ct_utils:shuffle(ConMiners)),
+    PayeeAddr = miner_ct_utils:node2addr(Payee, AddrList),
+    {ok, _Pubkey, SigFun, _ECDHFun} = ct_rpc:call(Miner, blockchain_swarm, keys, []),
+
+    StartNonce = miner_ct_utils:get_nonce(Miner, Addr),
+    {ok, _Pubkey, SigFun, _ECDHFun} = ct_rpc:call(Miner, blockchain_swarm, keys, []),
+
+    %% the first txn
+    Txn1 = ct_rpc:call(Miner, blockchain_txn_payment_v1, new, [PayerAddr, PayeeAddr, 1000, StartNonce+1]),
+    SignedTxn1 = ct_rpc:call(Miner, blockchain_txn_payment_v1, sign, [Txn1, SigFun]),
+    %% the second txn - with the same nonce as the first
+    Txn2 = ct_rpc:call(Miner, blockchain_txn_payment_v1, new, [PayerAddr, PayeeAddr, 1000, StartNonce+1]),
+    SignedTxn2 = ct_rpc:call(Miner, blockchain_txn_payment_v1, sign, [Txn2, SigFun]),
```

Comment:
This is an unused variable, I'm assuming you meant to use it down on line 437?

Reply:
well spotted! fixed...